[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling.

etaf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #173779
* #161940
* #161939
* #161938
* #160729
* #160706
* __->__ #160688


Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

etaf committed:
[Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling.

[ghstack-poisoned]

etaf committed:
Update on " [Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."




cc jeffdaily sunway513 jithunnair-amd pruthvistony ROCmSupport dllehr-amd jataylo hongxiayang naromero77amd voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."




cc jeffdaily sunway513 jithunnair-amd pruthvistony ROCmSupport dllehr-amd jataylo hongxiayang naromero77amd voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."




cc jeffdaily sunway513 jithunnair-amd pruthvistony ROCmSupport dllehr-amd jataylo hongxiayang naromero77amd voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."




cc jeffdaily sunway513 jithunnair-amd pruthvistony ROCmSupport dllehr-amd jataylo hongxiayang naromero77amd voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."






[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 6/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."






[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."






[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

EikanWang commented on torch/_inductor/codegen/combined_scheduling.py:
Why does this PR need to rename this file? Per my understanding, it is cuda-specific, and we just need to refine it by using `CUTLASSScheduling`

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACombinedScheduling` as `CombinedScheduling` and `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf commented:
Hi, @EikanWang This PR is ready for review, would you like to take a look again?

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 5/N: Refactor CUDACombinedScheduling and  CUDACppScheduling."



Following the design of #160175, This PR renamed `CUDACppScheduling` to `CUTLASSScheduling`  so that they can be reused for XPU.

cc jeffdaily sunway513 jithunnair-amd pruthvistony ROCmSupport jataylo hongxiayang naromero77amd pragupta jerrymannil xinyazhang voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]
